rotkehlchen/chain/ethereum/modules/makerdao/decoder.py

```diff
@@ -436,6 +443,68 @@ def decode_cdp_manager_events(self, context: DecoderContext) -> DecodingOutput:
 
         return DEFAULT_DECODING_OUTPUT
 
+    def decode_sdai_events(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] == SDAI_DEPOSIT:
+            # DAI -> SDAI: owner receives sdai
+            owner_address = self._get_address_or_proxy(hex_or_bytes_to_address(context.tx_log.topics[2]))  # noqa: E501
+            to_address = owner_address
+            from_address: Optional[ChecksumEvmAddress] = self.sdai.evm_address
+        elif context.tx_log.topics[0] == SDAI_REDEEM:
+            # SDAI -> DAI: owner deposits sdai
+            owner_address = self._get_address_or_proxy(hex_or_bytes_to_address(context.tx_log.topics[3]))  # noqa: E501
+            from_address = owner_address
+            to_address = self.sdai.evm_address
+        else:
+            return DEFAULT_DECODING_OUTPUT
+
+        if from_address is None or to_address is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(context.tx_log.data[32:64]),
+            asset=self.sdai,
+        )
+
+        if amount == ZERO:
+            return DEFAULT_DECODING_OUTPUT
+
+        if owner_address == from_address:
+            event_type = HistoryEventType.DEPOSIT
+            event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+            notes = f'Deposit {amount} {self.sdai.symbol} to sDAI Pot'
+        else:
+            event_type = HistoryEventType.WITHDRAWAL
+            event_subtype = HistoryEventSubType.REMOVE_ASSET
+            notes = f'Withdraw {amount} {self.sdai.symbol} from sDAI Pot'
+
+        transfer = self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=event_type,
+            event_subtype=event_subtype,
+            asset=A_SDAI,
+            balance=Balance(amount=amount),
+            location_label=owner_address,
+            notes=notes,
+            address=self.sdai.evm_address,
+            counterparty=None,
+            extra_data=None,
+        )
+
+        for event in context.decoded_events:
+            if event.event_subtype == HistoryEventSubType.NONE:
+                incoming_event = event.event_type == HistoryEventType.RECEIVE
+                event.asset = A_DAI
+                event.event_type = HistoryEventType.WITHDRAWAL if incoming_event else HistoryEventType.DEPOSIT  # noqa: E501
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET if incoming_event else HistoryEventSubType.DEPOSIT_ASSET  # noqa: E501
+                verb, preposition = ('Withdraw', 'from') if incoming_event else ('Deposit', 'to')
+                event.notes = f'{verb} {event.balance.amount} {self.dai.symbol} {preposition} sDAI Pot'  # noqa: E501
+
+        return DecodingOutput(
```

Comment:
you also need to make sure the order is as I wrote in the test. We use `maybe_reshuffle_events` for this.